hypothesis-python/src/hypothesis/internal/conjecture/shrinker.py

```diff
@@ -1455,33 +1455,33 @@ def block_program(description):
     running a block program its score will be updated.
     """
 
-    def run(self):
+    def run(self, i):
```

Comment:
It might be helpful to take the lambda that defines `i`, and hoist it up above this line so that it's easier to see the connection between the two.

(That would be more consistent with the structure of the other pass definitions, even though the `defines_shrink_pass` call has to stay where it is.)